plugins/mumble_plugin_utils.h

```diff
@@ -49,4 +53,38 @@ static void escape(char *str, size_t size) {
     }
 }
 
+// Calculates sine and cosine of the specified value.
+// On Linux the calculation is guaranteed to be simultaneous.
+static inline bool sinCos(const float &value, float &outSin, float &outCos) {
+#ifdef OS_WINDOWS
+	outSin = sin(value);
+	outCos = cos(value);
+	return true;
+#else
+	errno = 0;
+	feclearexcept(FE_ALL_EXCEPT);
+
+	sincosf(value, &outSin, &outCos);
+
+	return fetestexcept(FE_INVALID | FE_DIVBYZERO | FE_OVERFLOW | FE_UNDERFLOW) == 0;
+#endif
+}
+
+// Converts degrees to radians.
+static inline float degreesToRadians(const float &degrees) {
+	static const float piOver180 = M_PI / 180.0f;
+	return piOver180 * degrees;
+}
+
+// Converts between little endian and big endian.
+static inline uint16_t swapBits(const uint16_t &value) {
+	uint16_t ret;
+
+	// We split the variable(s) into an array of 2 uint8_t, to read/write the single bits.
```

Comment:
[`__builtin_bswap16`](https://gcc.gnu.org/onlinedocs/gcc/Other-Builtins.html) and [`_byteswap_ushort`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/byteswap-uint64-byteswap-ulong-byteswap-ushort) are probably better.